lms/djangoapps/course_home_api/progress/v1/views.py

```diff
@@ -171,19 +173,21 @@ def get(self, request, *args, **kwargs):
         data = {
             'end': course.end,
             'user_has_passing_grade': user_has_passing_grade,
-            'certificate_data': get_cert_data(request.user, course, enrollment_mode, course_grade),
+            'certificate_data': get_cert_data(request.user, course, enrollment.mode, course_grade),
             'completion_summary': get_course_blocks_completion_summary(course_key, request.user),
             'course_grade': course_grade,
             'has_scheduled_content': has_scheduled_content,
             'section_scores': course_grade.chapter_grades.values(),
-            'enrollment_mode': enrollment_mode,
+            'enrollment_mode': enrollment.mode,
             'grading_policy': grading_policy,
             'studio_url': get_studio_url(course, 'settings/grading'),
             'verification_data': verification_data,
         }
         context = self.get_serializer_context()
         context['staff_access'] = bool(has_access(request.user, 'staff', course))
         context['course_key'] = course_key
+        context['course_overview'] = course_overview
+        context['enrollment'] = enrollment
```

Comment:
I know this extends beyond your changes here, but I'm starting to wonder if we should document why/how these different fields are used/are necessary. Like coming into here in a year, it might be confusing why we are passing in these objects and maybe a little difficult to dig into the serializer and then the parent serializer to see it is required as part of that.
All of this is to say, I think I'd appreciate a comment here to say these two things are used for the `VerifiedModeSerializerMixin` and if you want to do any discovery on the other pieces of context, I wouldn't hate it.

Reply:
I agree, I'll add a comment here and see if the other context items could benefit from comments.